tests/amr/data/field/refine/test_refine_field.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+
+from pyphare.core import box as boxm
+from pyphare.core.gridlayout import GridLayout
+from pyphare.pharesee.hierarchy import FieldData
+
+
+# only supports ndim == 1, updates needed for > 1
+# assumes refinement_ratio is 2
+
+def refine(field, **kwargs):
+    assert isinstance(field, FieldData)
+    kwarg_keys = ["data"] # doing if data == None for input data=None, fails if it's an np array
+    assert any([key in kwarg_keys for key in list(kwargs.keys())])
+
+    refinement_ratio = 2
+    primal_directions = field.primal_directions()
+    data   = kwargs.get("data", field.dataset[:])
```

Comment:
at this point, it's not obvious to guess what's the goal of "data", in particular since its default value is the field's data. The function refines "data", which may come from kwargs or from the field's dataset, isn't it a bit weird? Why would you need to pass a field but don't use its data

Reply:
the field is for the other values, the "data" kwarg is an override in case you want to use another dataset, like in this PR, the time interpolated dataset

Reply:
and it's not possible to "wrap" the refined/time interpolated dataset into a FieldData based on the original, so to have a single data source?

Reply:
it is possible, seems like overkill to make a temporary field just to refine different data